Makefile

```diff
@@ -90,6 +110,80 @@ test-silent:
 benchmark-test:
 	. ./scripts/shared_env && go test -run=XX -bench=. $(shell go list ./agent/... | grep -v /vendor/)
 
+define dockerbuild
+	docker run \
+		--net none \
+		--user "$(USERID)" \
+		--volume "$(PWD)/out:/out" \
+		--volume "$(PWD):/go/src/github.com/aws/amazon-ecs-agent" \
+		--rm \
+		$(BUILDER_IMAGE) \
+		$(1)
+endef
+
+define win-cgo-dockerbuild
+	docker run --net=none \
+		--user "$(USERID)" \
+		--volume "$(PWD)/out:/out" \
+		--volume "$(PWD):/go/src/github.com/aws/amazon-ecs-agent" \
+		--env "GOOS=windows" \
+		--env "CGO_ENABLED=1" \
+		--env "CC=x86_64-w64-mingw32-gcc" \
+		--rm \
+		$(BUILDER_IMAGE) \
+		$(1)
+endef
+
+# TODO: use `go list -f` to target the test files more directly
+ALL_GO_FILES = $(shell find . -name "*.go" -print | tr "\n" " ")
+GO_INTEG_TEST = go test -race -tags integration -c -o
+out/test-artifacts/linux-engine-tests: out $(ALL_GO_FILES)
```

Comment:
All of these `out/` targets are deliberately not `PHONY` targets, yeah?

Reply:
They're not `.PHONY` because they're producing real files that make is then able to track.

Reply:
right, wanted to confirm that, thanks